### PR TITLE
use fields to avoid hard coded state classes

### PR DESCRIPTION
**Pull request by kennywangjin**
Summary of the changes (Less than 80 chars)
- add static read-write fields to EditContext to hold validation state classes
- use these classes to apply field validation state
- user could customize their own classes

### CODE REVIEW

File: src/Components/Components/src/Forms/EditContext.cs [hunk 1]
```diff
@@ -13,6 +13,12 @@ namespace Microsoft.AspNetCore.Components.Forms
     /// </summary>
     public sealed class EditContext
     {
+        public static string ModifiedClass = "modified";
```
Comment (gfoidl) [inline src/Components/Components/src/Forms/EditContext.cs]: Why don't you use constants?
And do they have to be `public`? `private` would be enough, as long no use outside this class is planned.

BTW: If you use `static readonly` the JIT can treat them es constants. Anyway, here `const` would be the better way.

Comment (SteveSandersonMS) [reply src/Components/Components/src/Forms/EditContext.cs]: @gfoidl The point of @kennywangjin's PR here is to make these values mutable, hence not `const`.

I'm not sure we'll use this approach, but hopefully it explains why the PR is as it is. It's quite likely that we'd move the responsibility for this into `EditForm` and have it configurable on a per-instance basis instead.

Comment (gfoidl) [reply src/Components/Components/src/Forms/EditContext.cs]: > hopefully it explains why the PR is as it is

Yes, thanks! 
(Unfortunately there's no reference to any issue, etc. in the description of the PR, and the boilerplate "Summary of the changes (Less than 80 chars)" distracted me.)

> have it configurable on a per-instance basis instead

Seems reasonable.

**Comment by mkArtakMSFT** [comment]
Thanks for your effort. We won't be accepting your change as is, as global static properties would allow breaking this from some other place.

**Pull request closed without merging**